syntax = "proto3";

option go_package = "github.com/y-akahori-ramen/gRPCUE4Demo/proto";
package grpcDemo;

message GetServerStreamingTextureRequest {
    string imageName = 1;
}

message GetServerStreamingTextureResponse {
    bytes chunk = 1;
}

message GetServerStreamingMessageRequest {
    int32 count = 1;
}

message GetServerStreamingMessageResponse {
    string message = 1;
}

service ServerStreamingDemoService {
    rpc GetTexture (GetServerStreamingTextureRequest) returns (stream GetServerStreamingTextureResponse) {}
    rpc GetSimpleMessage(GetServerStreamingMessageRequest) returns (stream GetServerStreamingMessageResponse) {}
}